packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -176,39 +178,57 @@ export const AlephiumConnectProvider: React.FC<{
   )
 
   useEffect(() => {
-    const func = async () => {
-      const onDisconnected = () => {
-        removeLastConnectedAccount()
-        updateAccount(undefined)
-        updateSignerProvider(undefined)
-      }
-      const onConnected = (result: ConnectResult) => {
-        updateAccount(result.account)
-        updateSignerProvider(result.signerProvider)
+    const onDisconnected = () => {
+      removeLastConnectedAccount()
+      updateAccount(undefined)
+      updateSignerProvider(undefined)
+    }
+    const onConnected = (result: ConnectResult) => {
+      updateAccount(result.account)
+      updateSignerProvider(result.signerProvider)
+    }
+
+    const reconnectByInjected = async (providers: AlephiumWindowObject[], times = 0) => {
+      if (times >= 20) {
```

Comment:
The reason I made this change is that our extension wallet injects the inpage provider using [a <script> tag](https://github.com/alephium/extension-wallet/blob/3b679e5b674dc6b6de11147bc1326a3d48f9da4f/packages/extension/src/content.ts#L9), which may run asynchronously in chrome manifest v3 due to [this issue](https://issues.chromium.org/issues/41267807). As a result, the dApp sends a `requestProvider` request but may not immediately receive a response. I tested `react-dapp-template`, `nextjs-app-dapp-template`, and bridge ui. Only bridge ui has this issue.

In the ETH ecosystem, detecting injected providers works like this:

1. The extension wallet calls `window.addEventListener('requestProvider', () => window.dispatchEvent('announceProvider', provider))` to handle the dApp's `requestProvider` requests.
2. The dApp calls `window.dispatchEvent('requestProvider')` to request providers.

A crucial assumption is that step 1 must be completed before step 2 begins. Otherwise, the dApp may not receive the provider, and `wagmi` is based on this assumption.

I checked the OneKey extension wallet and the MetaMask extension wallet:

1. OneKey wallet places [`injected.js` in `content_scripts`](https://github.com/OneKeyHQ/app-monorepo/blob/503c27c66f5dc47ce56a8aeee1b709bf17e846d9/apps/ext/src/manifest/chrome_v3.js#L97). In `injected.js`, OneKey [registers the wallet provider for each chain](https://github.com/OneKeyHQ/cross-inpage-provider/blob/c17620a963ca9d09d4b1c7d739411baf66db2b0f/packages/injected/src/injectedExtension.ts#L16), which ensures that `injected.js` completed before the dApp starts.
2. Just like our extension wallet, MetaMask used to inject the in-page provider using the `<script>` tag before. However, due to the chrome bug mentioned above, it now uses `chrome.scripting.registerContentScripts` instead in [this PR](https://github.com/MetaMask/metamask-extension/pull/15448).

I tried using the approach from OneKey and MetaMask to modify our extension wallet, but it didn't work as expected. I'm not sure why yet, perhaps it needs some other related code changes. So I've fixed the issue in the SDK, and it can work properly in different environments.

Reply:
Could you check how this is resolved in Argent X's latest code?

Reply:
I just checked the [starknetkit]()https://github.com/argentlabs/starknetkit code; unfortunately, they haven't addressed this issue. Starknetkit tries to load injected providers when the dApp starts, but due to the issue I described above, it sometimes fails to detect the installed Argent-X wallet. Here's a simple video demo where you can see that I have the Argent-X wallet installed, but when I refresh the page, there's a high chance that it will prompt me again to install the Argent-X wallet:

https://github.com/user-attachments/assets/dffbba8c-4a9e-4892-8adb-7479bdcf1f90

I'm not sure why no one has raised this issue with starknetkit. One possible reason is that they haven't tested with create-react-app. In the environments I've tested, this issue only occurs when using create-react-app. It never happens with vite or nextjs.

Reply:
I tried using the MetaMask approach, and it works now. So we can remove the last commit of this PR.

Do you guys think we should fix this issue in the extension wallet? If so, I will test it in different browsers later and create a PR to update the extension wallet. cc @h0ngcha0

Reply:
Yes, please fix the extension wallet accordingly! Thanks